pulsar-broker/src/main/java/org/apache/pulsar/broker/service/SystemTopicBasedTopicPoliciesService.java

```diff
@@ -526,32 +533,41 @@ private void initPolicesCache(SystemTopicClient.Reader<PulsarEvent> reader, Comp
         });
     }
 
-    private void cleanCacheAndCloseReader(@NonNull NamespaceName namespace, boolean cleanOwnedBundlesCount) {
-        cleanCacheAndCloseReader(namespace, cleanOwnedBundlesCount, false);
+    private void cleanCache(@NonNull NamespaceName namespace, boolean cleanOwnedBundlesCount,
+                            boolean closeReaderCache) {
+        cleanCache(namespace, cleanOwnedBundlesCount, false, closeReaderCache);
     }
 
-    private void cleanCacheAndCloseReader(@NonNull NamespaceName namespace, boolean cleanOwnedBundlesCount,
-                                          boolean cleanWriterCache) {
+    private void cleanCache(@NonNull NamespaceName namespace, boolean cleanOwnedBundlesCount,
+                            boolean cleanWriterCache, boolean cleanReaderCache) {
         if (cleanWriterCache) {
             writerCaches.synchronous().invalidate(namespace);
         }
-        CompletableFuture<SystemTopicClient.Reader<PulsarEvent>> readerFuture = readerCaches.remove(namespace);
 
         if (cleanOwnedBundlesCount) {
             ownedBundlesCountPerNamespace.remove(namespace);
         }
-        if (readerFuture != null && !readerFuture.isCompletedExceptionally()) {
-            readerFuture.thenCompose(SystemTopicClient.Reader::closeAsync)
-                    .exceptionally(ex -> {
-                        log.warn("[{}] Close change_event reader fail.", namespace, ex);
-                        return null;
-                    });
+
+        if (!cleanReaderCache) {
+            return;
         }
 
+        CompletableFuture<SystemTopicClient.Reader<PulsarEvent>> readerFuture = readerCaches.remove(namespace);
         policyCacheInitMap.compute(namespace, (k, v) -> {
             policiesCache.entrySet().removeIf(entry -> Objects.equals(entry.getKey().getNamespaceObject(), namespace));
             return null;
         });
+        if (readerFuture != null && !readerFuture.isCompletedExceptionally()) {
+            readerFuture
+                    .thenCompose(r -> {
+                        log.info("[{}] Closing the topic policies reader.", namespace);
```

Comment:
This extra log might be redundant.  There is a `Closed consumer` log here when closing: https://github.com/apache/pulsar/blob/f2709bc823c7d048065dfb3d2845e47bcb0e3e0f/pulsar-client/src/main/java/org/apache/pulsar/client/impl/ConsumerImpl.java#L1221